sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/ObjectSerializer.java

```diff
@@ -22,28 +40,69 @@
      */
     <T> T deserialize(InputStream stream, TypeReference<T> typeReference);
 
+    /**
+     * Reads a byte array into its object representation.
+     *
```

Comment:
javadoc : Adding behavior that providing `null` as `data` will return `Mono.empty()`

Reply:
I'm not certain that I want to add that to the interface Javadocs as an implementation could change that detail. For our implementations we return `null` but others may wish to return a default value.

Reply:
Updated this to pass `new byte[0]` and allow the implementation to handle deserialization.